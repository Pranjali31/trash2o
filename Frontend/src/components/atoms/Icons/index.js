import React from 'react';
import {Icon} from '@rneui/themed';
import Colors from '../../../theme/colors';

const Icons = ({name, type, color, size}) => {
  return (
    <Icon
      name={name}
      type={type}
      color={color ? color : Colors.primaryColor}
      size={size ? size : 20}
    />
  );
};

export default Icons;
