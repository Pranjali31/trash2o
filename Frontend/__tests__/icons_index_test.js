import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import {Icon} from '@rneui/themed';
import Icons from '../src/components/atoms/Icons/index'

test("index Component Loading",()=>{
    render(<Icons name="welcome"  type="button" color={null} size={null} />)
 
 })
 test("index component snapshot", () => {
    let component = render(<Icons name="welcome"  type="button" color={null} size={null}/>).toJSON();
    //console.log(JSON.stringify(component))
    expect(component).toMatchSnapshot();
})


describe("index Testing component", () => {
    var component;
    beforeEach(() => {
        component = render(<Icons name="welcome"  type="button" color={null} size={null}/>).toJSON();

    });

    test("Testing static text data", () => {
        let styles = component.children[0].children[0].props.style;
        expect(styles).toStrictEqual({
            backgroundColor	:	"transparent",
            alignItems	: "center",
            justifyContent	: "center"

        });
    });

})