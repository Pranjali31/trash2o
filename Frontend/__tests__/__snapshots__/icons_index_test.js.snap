// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index component snapshot 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
    }
  }
  testID="RNE__ICON__CONTAINER"
>
  <View
    accessibilityRole="button"
    android_ripple={
      Object {
        "borderless": false,
        "color": "rgba(0, 0, 0, 0.3)",
        "radius": -5,
      }
    }
    disabled={false}
    testID="RNE__ICON__CONTAINER_ACTION"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      testID="RNE__ICON"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#0D110F",
              "fontSize": 20,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="RNE__ICON__Component"
      >
        ?
      </Text>
    </View>
  </View>
</View>
`;
